### ISSUE TITLE
Training Early Stopping.

### ISSUE THREAD

**Issue by Voldy1998** [AntonMu/TrainYourOwnYOLO#95]
Before filing a report consider the following two questions:

### Have you checked the [troubleshooting](https://github.com/AntonMu/TrainYourOwnYOLO#troubleshooting) section? YES

Once you are familiar with the code, you're welcome to modify it. Please only continue to file a bug report if you encounter an issue with the provided code and after having followed the instructions.

If you have followed the instructions exactly, couldn't solve your problem with the provided troubleshooting tips and would still like to file a bug or make a feature requests please follow the steps below.

1. It must be a bug, a feature request, or a significant problem with the documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.

------------------------

### System information
- **What is the top-level directory of the model you are using**: /content/drive/My Drive/TrainYourOwnYOLO/2_Training. This is done in google colab since my gpu memory is not sufficient.
- **Have I written custom code (as opposed to using a stock example script provided in the repo)**: NO
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **TensorFlow version (use command below)**: 1.15.0
- **CUDA/cuDNN version**:  Trained using Google Colab
- **GPU model and memory**: Trained using colab GPU
- **Exact command to reproduce**:  python Train_YOLO.py

You can obtain the TensorFlow version with

`python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug or a feature request.
Previously I had trained using 400 images of cars and the training stopped at the 78/102 epoch at loss of 13.1176 and valid loss of 16.9962. So as per your suggestion this time I trained using 1000 images. But this time it is stopping at 71st epoch at loss: 15.9903 - val_loss: 16.2303. Images have been attached. I did try testing it with images of cars but the results are not perfect.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.
Epoch 00070: ReduceLROnPlateau reducing learning rate to 1.0000000116860975e-08.
Epoch 71/102
225/225 [==============================] - 89s 395ms/step - loss: 15.9903 - val_loss: 16.2303
Epoch 00071: early stopping

![earlystopping](https://user-images.githubusercontent.com/40512980/75622584-042f0a80-5bc8-11ea-9eff-598817271df9.GIF)
The detection results on test images are as shown below:
![krukru_catface](https://user-images.githubusercontent.com/40512980/75622606-49ebd300-5bc8-11ea-9be2-93516a73609f.jpg)
In this image that red car in the side is not detected.
![backedintoparking-1a_catface](https://user-images.githubusercontent.com/40512980/75622611-65ef7480-5bc8-11ea-80a6-5ca614fe2cfb.jpg)
In this image many of the cars are not getting detected.

What can be done to improve the detection accuracy?

**Comment by AntonMu** [comment]
Hi @Voldy1998 - it looks that there are simply too many objects in one picture. YOLO uses anchors and region proposals and there is a limit of how many objects it can detect. Remember also that each image is downsampled to 416x416 pixels and thus your cars become very small in such a large picture. 

I recommend splitting up your big images into several parts (for example turn one big picture into 4 or 9 smaller pictures). You should do this both for training and for inference.

Hope this helps!

**Comment by Voldy1998** [comment]
Thanks for the suggestion @AntonMu . I have another query.
![antonmu](https://user-images.githubusercontent.com/40512980/76489616-ec207c00-644e-11ea-9d1e-e308ba32ba83.GIF)

I have tested the model using 101 images. But each time i run that piece of code "python Detector.py", time taken to process 101 images (for the same set of images) changes. Sometimes it increases and sometimes it decreases. Yesterday I got 53 sec now it's 125.4 sec. What could be the reason for it?
Also we can see that to process a video it takes 110.3 sec (even this time is not fixed). How to reduce time taken to process a video?

**Comment by Voldy1998** [comment]
> Hi @Voldy1998 - it looks that there are simply too many objects in one picture. YOLO uses anchors and region proposals and there is a limit of how many objects it can detect. Remember also that each image is downsampled to 416x416 pixels and thus your cars become very small in such a large picture.
> 
> I recommend splitting up your big images into several parts (for example turn one big picture into 4 or 9 smaller pictures). You should do this both for training and for inference.
> 
> Hope this helps!

I am tried using the pre-trained yolo model to detect cars in the below image.
![withoutseg](https://user-images.githubusercontent.com/40512980/82323946-6b55a180-99f6-11ea-81e2-624c32eebe0f.jpeg)
But it couldn't detect the cars which are smaller and in the background.
So I tried segmenting the image into multiple blocks and gave it to the pre-trained model.
![WhatsApp Image 2020-05-14 at 9 22 43 PM](https://user-images.githubusercontent.com/40512980/82324025-93450500-99f6-11ea-9d32-e5bcc0149c7a.jpeg)
![aaaa](https://user-images.githubusercontent.com/40512980/82324468-50cff800-99f7-11ea-804c-1f4c781afc31.jpg)

As you can see there isn't much improvement. What can be done so that the pre-trained yolo model can detect all the cars in the image?
